tests/console/validate_packages_and_patterns.pm

```diff
@@ -64,7 +64,7 @@ elsif (check_var('VALIDATE_PCM_PATTERN', 'aws')) {
     # Define more packages with the same set of expectations
     $software{'update-test-interactive'} = $software{'update-test-feature'};
     $software{'update-test-security'}    = $software{'update-test-feature'};
-    $software{'update-test-trival'}      = $software{'update-test-feature'};
+    $software{'update-test-trivial'}     = $software{'update-test-feature'};
```

Comment:
I believe it would be cleaner to define expectations depending on the version here instead of evaluating 3-4 conditions on each loop iteration. WDYT?

Reply:
Yes, but knowing the conditions, you can set expectations accordingly, introducing that hack in the loop is less readable, is not efficient from the performance perspective. As for `AC3: Other product versions which have the typo-fixed version also work`, there is already a mechanism to perform dynamic checks pasting code reference to `condition` key, where you can check which package is installed.

Reply:
Updated patch as we discussed.